### PR TITLE
Decode polyline6

### PR DESCRIPTION
**Pull request by frederoni**
We're decoding a geometry string as `polyline` even if we specify `polyline6` in the route options.

@bsudekum @1ec5

### CODE REVIEW

File: MapboxDirections/MBRoute.swift [hunk 1]
```diff
@@ -38,7 +38,8 @@ open class Route: DirectionsResult {
         case let geometry as JSONDictionary:
             coordinates = CLLocationCoordinate2D.coordinates(geoJSON: geometry)
         case let geometry as String:
-            coordinates = decodePolyline(geometry, precision: 1e5)!
+            let precision = routeOptions.shapeFormat == .polyline ? 1e5 : 1e6
```
Comment (frederoni) [inline MapboxDirections/MBRoute.swift]: Probably better to switch on `routeOptions.shapeFormat`?

Comment (bsudekum) [reply MapboxDirections/MBRoute.swift]: Yeah, probably. Although the line above, `case let geometry as String:` takes care of the case where it's geojson.

Comment (frederoni) [reply MapboxDirections/MBRoute.swift]: We need to plumb this through `MBRouteStep` as well. 🤔

**Comment by frederoni** [comment]
Ready for review

**Pull request merged**